packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ButtonHTMLAttributes, Fragment, HTMLAttributes, memo, ReactNode, useMemo } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown, DropdownProps } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface ButtonGroupItem extends ButtonHTMLAttributes<HTMLButtonElement> {
+  content: string | ReactNode;
+}
+
+export interface ButtonGroupDropdownItem extends Omit<DropdownProps, 'toggle'> {
+  toggle: ButtonGroupItem;
+}
```

Comment:
We should do this overflow dropdown for the consumer. Essentially get rid of this interface and just make the component consume an array of `actions`. This overflow button could reuse some of the logic we did for the [`PillTabs` component](https://github.com/bigcommerce/big-design/blob/master/packages/big-design/src/components/PillTabs/PillTabs.tsx#L34). I would consider abstracting some of that overflow logic into a utility hook or something and reuse it in both places.

Reply:
I 100% agree with this

Reply:
Got it, thank you, guys!

I have seen the `PillTabs` and I thought that the `ButtonGroup` concept should be a lil bit different but, yeah, I agree with you and re-build my version 🤝 

cc @jorgemoya @chanceaclark